toolchain/parse/testdata/let/fail_associated_const_default_initializer.carbon

```diff
@@ -0,0 +1,32 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/parse/testdata/let/fail_associated_const_default_initializer.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/parse/testdata/let/fail_associated_const_default_initializer.carbon
+
+interface MagicBox {
+  // CHECK:STDERR: fail_associated_const_default_initializer.carbon:[[@LINE+4]]:30: error: `let` declarations must end with a `;` [ExpectedDeclSemi]
+  // CHECK:STDERR:   let secretMessage:! String = "This is a super secret";
```

Comment:
As the issue mentions,

> the pattern must be a single symbolic binding pattern (i.e. "name :! type"), and it can't have an initializer (we don't support default let at all yet)

I'm thinking what could be the correct diagnostic for this case 🤔 @geoffromer.